dev_guide/templates.adoc

```diff
@@ -609,6 +632,85 @@ namespace the parameter substitution resolved the value to, assuming the user
 has permission to create objects in that namespace.
 ====
 
+[[writing-exposing-object-fields]]
+
+=== Exposing Object Fields
+
+Template authors may indicate that fields of particular objects in a template
+should be exposed. The template service broker recognises exposed fields on
+ConfigMap, Secret, Service and Route objects and returns the values of the
+exposed fields when a user binds a service backed by the broker.
+
+To expose one or more fields of an object, add annotations prefixed by
+`template.openshift.io/expose-` or `template.openshift.io/base64-expose-` to
+the object in the template. The annotation key following the prefix is passed
+through to key the returned field in the bind response.
+
+Use the `template.openshift.io/expose-` annotation to return the field value as
+a string. This is convenient although it does not handle arbitrary binary data.
+If you wish to return binary data, the `template.openshift.io/base64-expose-`
```

Comment:
use the the `template.openshift.io/base64-expose-` annotation instead to base64 encode the data before it is returned.